napari/layers/utils/_slice_input.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import annotations
+
+import warnings
+from dataclasses import dataclass
+from typing import List, Tuple, Union
+
+import numpy as np
+
+from napari.utils.transforms import Affine
+from napari.utils.translations import trans
+
+
+@dataclass(frozen=True)
+class _SliceInput:
+    """Encapsulates the input needed for slicing a layer.
+
+    An instance of this should be associated with a layer and some of the values
+    in ``Viewer.dims`` when slicing a layer.
+    """
+
+    # The number of dimensions displayed to be displayed in the slice.
+    ndisplay: int
```

Comment:
I don't have a strong opinion here because I don't know if the naming is following some convention. But our current naming is very generic and requires docs to note their semantics. Some variable renaming would be sweet

Reply:
The naming is mostly the existing naming. I'm unsure if it's following a convention outside of napari. Happy to take suggestions, but my main focus was to at least clarify what these things are with docstrings. If there are ways to make concise versions of those that can be directly reflected in the names, that would be great.

Reply:
For this refactor, I suggest to keep these names (so the rest of the diff looks as close as possible to how it was before), but I agree that we can make these better.